driver-sync/src/test/functional/com/mongodb/client/AbstractSessionsProseTest.java

```diff
@@ -71,7 +71,9 @@ public void shouldCreateServerSessionOnlyAfterConnectionCheckout() throws Interr
                         .addCommandListener(new CommandListener() {
                             @Override
                             public void commandStarted(final CommandStartedEvent event) {
-                                lsidSet.add(event.getCommand().getDocument("lsid"));
+                                if (event.getCommand().containsKey("lsid")) {
```

Comment:
What problem did this address? I can't think of a command that wouldn't have an lsid on it.

Reply:
`com.mongodb.reactivestreams.client.SessionsProseTest.shouldIgnoreImplicitSessionIfConnectionDoesNotSupportSessions` failed on 4.2, 5.0, 6.0 due to this erroring.

Reply:
That seems impossible, since this code change only effects the `shouldCreateServerSessionOnlyAfterConnectionCheckout` in which it is contained.  I think something else must be happening that caused `shouldIgnoreImplicitSessionIfConnectionDoesNotSupportSessions` test to fail.

Reply:
```
[2023/11/07 18:16:39.381] 18:13:54.022 [Test worker] WARN  org.mongodb.driver.protocol.event - Exception thrown raising command started event to listener com.mongodb.client.AbstractSessionsProseTest$1@36ad5f2a
[2023/11/07 18:16:39.381] org.bson.BsonInvalidOperationException: Document does not contain key lsid
[2023/11/07 18:16:39.381] 	at org.bson.BsonDocument.throwIfKeyAbsent(BsonDocument.java:883) ~[bson-4.12.0-SNAPSHOT.jar:na]
[2023/11/07 18:16:39.381] 	at org.bson.BsonDocument.getDocument(BsonDocument.java:153) ~[bson-4.12.0-SNAPSHOT.jar:na]
[2023/11/07 18:16:39.381] 	at com.mongodb.client.AbstractSessionsProseTest$1.commandStarted(AbstractSessionsProseTest.java:74) ~[test/:na]
[2023/11/07 18:16:39.381] 	at com.mongodb.internal.connection.ProtocolHelper.sendCommandStartedEvent(ProtocolHelper.java:283) ~[mongodb-driver-core-4.12.0-SNAPSHOT.jar:na]
[2023/11/07 18:16:39.381] 	
```

However, looks like there was also an exception opening socket. Which maybe the root cause.

https://parsley.mongodb.com/test/mongo_java_driver_tests_jdk8_unsecure__version~4.2_os~linux_topology~standalone_auth~noauth_ssl~nossl_jdk~jdk8_test_patch_616ab9d32470d5f9aec3abdd3c3664256c49f52f_654a7c0cc9ec44fde9670c48_23_11_07_18_03_57/0/7b211511de7ffca6cd7363742c881993?bookmarks=0,247&shareLine=1

Reply:
Reverting